### ISSUE TITLE
Help for `dexit` displays non-existent `Subcommands` section

### ISSUE THREAD

**Issue by ErichDonGubler** [nushell/nushell#13278]
### Describe the bug

I think `How to reproduce` explains this sufficiently.

### How to reproduce

1. Run `dexit --help`. Observe that a `Subcommands` section that appears to be re-used from the `drop` built-in is displayed, which don't actually exist and are rejected if attempted to be used (i.e., `dexit nth` yields an `extra positional argument` error).

### Expected behavior

I expect the `dexit`/`std dirs drop` command to not have a help section from an unrelated command included.

### Screenshots

<img width="1173" alt="Screenshot 2024-06-30 at 7 42 27 PM" src="https://github.com/nushell/nushell/assets/658538/cb053902-46b2-4125-8138-7d3373fa6b7c">
<img width="434" alt="Screenshot 2024-06-30 at 7 58 31 PM" src="https://github.com/nushell/nushell/assets/658538/dcd485c8-41d4-45b8-9658-7c81384eea81">

### Configuration

| key                | value                                          |
| ------------------ | ---------------------------------------------- |
| version            | 0.95.0                                         |
| major              | 0                                              |
| minor              | 95                                             |
| patch              | 0                                              |
| branch             |                                                |
| commit_hash        |                                                |
| build_os           | macos-aarch64                                  |
| build_target       | aarch64-apple-darwin                           |
| rust_version       | rustc 1.79.0 (129f3b996 2024-06-10) (Homebrew) |
| cargo_version      | cargo 1.79.0                                   |
| build_time         | 2024-06-25 18:29:47 +00:00                     |
| build_rust_channel | release                                        |
| allocator          | mimalloc                                       |
| features           | default, sqlite, system-clipboard, trash       |
| installed_plugins  |                                                |

### Additional context

I'm guessing that this is a problem because there is overlap between the resolved name of `dexit` (viz., `std dirs drop`) and the name of the `drop` built-in. Assuming this is true, the lack of an `Examples` section that would normally be present with help from the `drop` built-in might be due to <https://github.com/nushell/nushell/issues/11067>?

**Comment by fdncred** [comment]
This is because `dexit` is just an alias to `drop` https://github.com/nushell/nushell/blob/e5cf4863e95126080439479328f55a26b9f65add/crates/nu-std/std/dirs.nu#L78. I agree that help is a little weird here.

**Comment by ErichDonGubler** [comment]
@fdncred: Right, the `std dirs drop` and `drop` built-in have the same "command basename", but are rather unrelated. I...think we're agreeing here, but I'm not sure? 😅

**Comment by fdncred** [comment]
I'm just explaining why help for dexit display drop because you're really doing help for drop. We could name the dirs drop command something else but it's still going to have the same problem that help dexit will still show help for something else.

**Comment by ErichDonGubler** [comment]
@fdncred: I intended this bug report to be about the `Subcommands` section from the _`drop` built-in_ that shouldn't be present at all in `dexit`'s help, rather than help content being correct but not explaining that, i.e., `dexit` corresponds to `std dirs drop`. Is there a way I could make that more clear?

**Comment by fdncred** [comment]
Right. The subcommands show up because you're asking for help on drop. That's the way it's supposed to work. If one asks for help on drop it shows the help and all subcommands that start with drop. This is how it works with str as well. I'm saying that we should've named drop for dexit differently so that it doesn't show up here at all nor does it reference the other drop commands.

The other strategy is to call the help based on the alias name versus the alias target's name.

**Comment by j-xella** [comment]
nushell `0.104.0` - the problem is still there...
